flink-batch-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormat.java

```diff
@@ -119,7 +118,11 @@ private boolean isSplitConfigured() {
 		return splitColumnName != null;
 	}
 
+	/** Perform initialization of the JDBC connection. Done just once per parallel task */
 	private void establishConnection() throws SQLException, ClassNotFoundException {
+		if(dbConn!=null){
```

Comment:
it just seems odd that we call this method n times only to do nothing in n-1 cases.